hdate/tekufot.py

```diff
@@ -0,0 +1,245 @@
+"""
+The class attempts to compute:
+    - The tekufot (seasons) in the Hebrew calendar: Nissan, Tammuz, Tishrei, and Tevet.
+    - Calculaltion are based on Shmuel calendar (Talmud Bavli, Eruvin 56a)
+    - Rav Ada’s Tekufah: Closer to the astronomical solar year,
+      shorter by 5min than Schmuel (365 days, 5 hours, 997 parts/chalakim) is not used
+    - Cheilat Geshamim start date, which differs between the diaspora and Israel.
+    - Halachic prayer periods based on key Jewish holidays and seasonal changes.
+    - Appropriate prayer phrases depending on the current date, tradition, and language.
+"""
+
+import datetime as dt
+from datetime import tzinfo
+from enum import Enum
+from typing import Dict, Union
+
+from hdate.hebrew_date import HebrewDate
+from hdate.htables import Months
+from hdate.location import Location
+from hdate.translator import TranslatorMixin
+from hdate.zmanim import Zmanim
+
+
+class Gevurot(TranslatorMixin, Enum):
+    """Enum class for the gevurot."""
+
+    MORID_HATAL = 0
+    MASHIV_HARUACH = 1
+    NEITHER = 2
+
+
+class Geshamim(TranslatorMixin, Enum):
+    """Enum class for the geshamim."""
+
+    BARKHEINU = 0
+    BARECH_ALEINU = 1
+    VETEN_TAL = 2
+    VETEN_BERACHA = 3
+
+
+class Tekufot(TranslatorMixin):  # pylint: disable=too-many-instance-attributes
+    """
+    A class that calculates and manages Jewish seasonal times (Tekufot),
+    periods for prayer insertions, and associated halachic dates such as
+    the start of Cheilat Geshamim (requesting rain)."""
+
+    def __init__(
+        self,
+        date: dt.date = dt.datetime.today(),
+        location: Location = Location(),
+        tradition: str = "israel",
+        language: str = "english",
+    ):
+        """Initialize the Tekufot object."""
+        super().__init__()
+
+        self.date = date
+        self.location = location
+        self.tradition = tradition or (
+            "israel" if not location.diaspora else "diaspora_sephardi"
+        )
+        self.language = language
+
+        # Convert current date Hebrew Date
+        self.hebrew_date = HebrewDate.from_gdate(date)
+        self.hebrew_year_p = self.hebrew_date.year
```

Comment:
I suggest you directly move to gregorian_year_p.
No need for the intermediate variable. Or else make it a local variable